engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/WorldSetupScreen.java

```diff
@@ -16,18 +16,213 @@
 
     @Override
     public void initialise() {
 
-        WidgetUtil.trySubscribe(this, "close", button ->
-                triggerBackAnimation()
-        );
+        WidgetUtil.trySubscribe(this, "apply", button -> {
+            triggerBackAnimation();
+        });
+
+        WidgetUtil.trySubscribe(this, "close", button -> {
+            world.setWorldConfigurator(oldWorldConfig);
+            triggerBackAnimation();
+        });
+    }
+
+    /**
+     * This method sets the world whose properties are to changed. This function is called before the screen comes
+     * to the forefront.
+     * @param subContext As there is a new environment created in the {@link UniverseSetupScreen}, it's passed to
+     *                   this screen instead of using the old Context.
+     * @param worldSelected the world whose configurations are to be changed.
+     * @throws UnresolvedWorldGeneratorException
+     */
+    public void setWorld(Context subContext, WorldSetupWrapper worldSelected) throws UnresolvedWorldGeneratorException {
+        world = worldSelected;
+        context = subContext;
+        SimpleUri worldGenUri = worldSelected.getWorldGeneratorInfo().getUri();
+        environment = context.get(ModuleEnvironment.class);
+        context.put(WorldGeneratorPluginLibrary.class, new TempWorldGeneratorPluginLibrary(environment, context));
+        if (world.getWorldGenerator() == null) {
+            worldGenerator = WorldGeneratorManager.createWorldGenerator(worldGenUri, context, environment);
+            world.setWorldGenerator(worldGenerator);
+        } else {
+            worldGenerator = world.getWorldGenerator();
+        }
+        //worldGenerator = WorldGeneratorManager.createWorldGenerator(worldGenUri, context, environment);
+        configureProperties();
+    }
+
+    /**
+     * Assigns a {@link WorldConfigurator} for every world if it doesn't exist. Using
+     * the WorldConfigurator it gets the properties associated with that world.
+     */
+    private void configureProperties() {
+
+        PropertyLayout propLayout = find("properties", PropertyLayout.class);
+        propLayout.setOrdering(PropertyOrdering.byLabel());
+        propLayout.clear();
+        WorldConfigurator worldConfig;
+        if (world.getWorldConfigurator() != null) {
+            worldConfig = world.getWorldConfigurator();
+        } else {
+            worldConfig = worldGenerator.getConfigurator();
+            world.setWorldConfigurator(worldConfig);
+        }
+        oldWorldConfig = worldConfig;
+
+        Map<String, Component> params = worldConfig.getProperties();
+
+        for (String key : params.keySet()) {
+            Class<? extends Component> clazz = params.get(key).getClass();
+            Component comp = config.getModuleConfig(worldGenerator.getUri(), key, clazz);
+            if (comp != null) {
+                worldConfig.setProperty(key, comp);       // use the data from the config instead of defaults
```

Comment:
Generally I try to put comments above a line instead of later on the line. That might be personal preference and situational however.